servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ResponseCancelTest.java

```diff
@@ -312,13 +337,19 @@ void resume() {
             }
         }
 
-        @SuppressWarnings("unchecked")
         static void resume(BlockingQueue<ClientTerminationSignal> signals,
-                           final CountDownLatch latch) throws Throwable {
-            ClientTerminationSignal signal = signals.take();
+                           int requestId,
+                           CountDownLatch latch) throws Throwable {
+            ClientTerminationSignal signal;
+            do {
+                // In case of cancel, a terminal signal may or may not arrive to the subscriber. The requestId helps
+                // to make sure we discard optional signals of all previous requests and resuming only for the current
+                // request.
+                signal = signals.take();
+            } while (signal.requestId != requestId);
```

Comment:
nit: would logging `if (signal.requestId != requestId)` provide any value here in case the test stalls?